tlatools/org.lamport.tlatools/src/tla2sany/parser/JListStack.java

```diff
@@ -0,0 +1,137 @@
+package tla2sany.parser;
+
+import java.util.Stack;
+
+/**
+ * Handling vertically-aligned conjunction & disjunction lists (called JLists
+ * for short) is one of the most difficult parts of parsing TLA+. This class
```

Comment:
I use the term extensively in the tree-sitter grammar (which is quite a bit more documented than the parsing code here so I got sick of writing "junction lists") but given the limited usage here it is probably fine to use the longer term junction lists instead of jlists.